File: Fiction/src/main/java/framework/utils/SPUtils.java
package framework.utils;

import android.annotation.TargetApi;
import android.content.Context;
import android.content.SharedPreferences;
import android.os.Build;

/**
 * by y on 2017/1/8.
 */

public class SPUtils {

    private static SharedPreferences sharedPreferences;
    private static final String SHAREDPREFERENCES_NAME = "fiction";

    @TargetApi(Build.VERSION_CODES.JELLY_BEAN)
    private static void initSharePreferences(Context context) {
        sharedPreferences = context.getSharedPreferences(SHAREDPREFERENCES_NAME, Context.MODE_PRIVATE);
    }

    public static void init(Context context) {
        initSharePreferences(context);
    }

    private static boolean isNull() {
        return sharedPreferences == null;
    }

    public static boolean getBoolean(String key, boolean defaultValue) {
        if (isNull()) {
            return defaultValue;
        }
        return sharedPreferences.getBoolean(key, defaultValue);
    }

    public static void setBoolean(String key, boolean value) {
        if (isNull()) {
            return;
        }
        sharedPreferences.edit().putBoolean(key, value).apply();
    }


    public static String getString(String key, String defaultValue) {
        if (isNull()) {
            return defaultValue;
        }
        return sharedPreferences.getString(key, defaultValue);
    }

    public static void setString(String key, String value) {
        if (isNull()) {
            return;
        }
        sharedPreferences.edit().putString(key, value).apply();
    }

    public static void clearAll() {
        if (isNull()) {
            return;
        }
        sharedPreferences.edit().clear().apply();
    }

}

File: sample_fuckapp/src/main/java/com/fuckapp/utils/SPUtils.kt
package com.fuckapp.utils

import android.content.Context
import android.content.SharedPreferences

/**
 * by y on 2016/10/20.
 */

object SPUtils {
    private lateinit var sharedPreferences: SharedPreferences
    private const val SHAREDPREFERENCES_NAME = "mokeeUninstallApp"

    const val WARN_APP = "WarnApp"

    private fun initSharePreferences(context: Context) {
        sharedPreferences =
            context.getSharedPreferences(SHAREDPREFERENCES_NAME, Context.MODE_PRIVATE)
    }

    fun init(context: Context) {
        initSharePreferences(context)
    }

    fun getString(key: String, defaultValue: String): String {
        return sharedPreferences.getString(key, defaultValue) ?: defaultValue
    }

    fun setString(key: String, value: String) {
        sharedPreferences.edit().putString(key, value).apply()
    }

    fun setBoolean(key: String, value: Boolean) {
        sharedPreferences.edit().putBoolean(key, value).apply()
    }

    fun getBoolean(key: String): Boolean {
        return sharedPreferences.getBoolean(key, false)
    }

    fun clearAll() {
        sharedPreferences.edit().clear().apply()
    }
}
